File: src/main/java/com/qtech/bubbles/common/ammo/AmmoType.java
package com.qtech.bubbles.common.ammo;

import com.qtech.bubbles.common.AttributeMap;
import com.qtech.bubbles.common.GraphicsProcessor;
import com.qtech.bubbles.common.RegistryEntry;
import com.qtech.bubbles.entity.AmmoEntity;

import java.awt.*;

public abstract class AmmoType extends RegistryEntry {
    public abstract void render(GraphicsProcessor g, AmmoEntity rotation);

    @SuppressWarnings("EmptyMethod")
    public void onCollision() {
        // Do nothing
    }

    public abstract AttributeMap getDefaultAttributes();

    public abstract Shape getShape(AmmoEntity ammoEntity);
}

File: src/main/kotlin/com/qtech/bubbles/common/ammo/AmmoType.kt
package com.qtech.bubbles.common.ammo

import com.qtech.bubbles.common.AttributeMap
import com.qtech.bubbles.common.RegistryEntry
import com.qtech.bubbles.entity.AmmoEntity
import com.qtech.bubbles.graphics.GraphicsProcessor
import java.awt.Shape

abstract class AmmoType : RegistryEntry() {
    abstract fun render(g: GraphicsProcessor, ammoEntity: AmmoEntity)
    fun onCollision() {
        // Do nothing
    }

    abstract val defaultAttributes: AttributeMap
    abstract fun getShape(ammoEntity: AmmoEntity): Shape
}